datastore/sync_entity.go

```diff
@@ -292,6 +294,12 @@ func (dynamo *Dynamo) UpdateSyncEntity(entity *SyncEntity) (bool, bool, error) {
 	}
 	if entity.Deleted != nil {
 		update = update.Set(expression.Name("Deleted"), expression.Value(entity.Deleted))
+
+		// Set TTL attribute when soft-deleting an item which will remove obsolete
+		// items automatically by dynamoDB.
+		if *entity.Deleted {
+			update = update.Set(expression.Name(ttlAttrName), expression.Value(time.Now().Add(ttl).Unix()))
+		}
```

Comment:
This means we will delete this forever, and will not be recoverable.  I would suggest we create a cold storage s3 bucket where these soft deleted expired items exist.  Or we need clear expectations to the users that soft delete means it will be deleted forever after 90 days.

Reply:
Great point, will create an issue and kick-off a follow up discussion on this with @jsecretan and @hspencer77, we could definitely explore more options regarding the actions we want to take after this expired on dynamo, code would stay the same for go-sync here so merging.